### PR TITLE
Implementing Series.tail() & DataFrame.tail()

### LINKED ISSUE #343: pandas.DataFrame.tail
Head is implemented but tail is not. Tail is not just useful for exploring data but also a very useful way of slicing data. 

Example: users have data ordered by date. They want to take the last 30 days. They can just call tail(30)

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.tail.html

### PR DESCRIPTION
**Pull request by itholic**
As since Spark 3.0 supports a `tail()`, this PR proposes a `Series.tail()` and `DataFrame.tail()` for Koalas based on it.

- Series
```python
>>> kser = ks.Series([1, 2, 3, 4, 5])
>>> kser
0    1
1    2
2    3
3    4
4    5
Name: 0, dtype: int64

>>> kser.tail(3)
2    3
3    4
4    5
Name: 0, dtype: int64
```

- DataFrame
```python
>>> df = ks.DataFrame({'animal': ['alligator', 'bee', 'falcon', 'lion',
...                    'monkey', 'parrot', 'shark', 'whale', 'zebra']})
>>> df
      animal
0  alligator
1        bee
2     falcon
3       lion
4     monkey
5     parrot
6      shark
7      whale
8      zebra

>>> df.tail()
   animal
4  monkey
5  parrot
6   shark
7   whale
8   zebra

>>> df.tail(3)
  animal
6  shark
7  whale
8  zebra

>>> df.tail(-3)
   animal
3    lion
4  monkey
5  parrot
6   shark
7   whale
8   zebra
```

Resolves #343

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -4992,6 +4992,77 @@ def items(self) -> Iterable:
         """This is an alias of ``iteritems``."""
         return self.iteritems()
 
+    def tail(self, n=5):
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py:4995]: @itholic can we implement it in DataFrame here too? I thought the implementation will be pretty similar

Comment (itholic) [reply databricks/koalas/series.py:4995]: @HyukjinKwon Sounds good!

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -10209,6 +10210,90 @@ def calculate_columns_axis(*cols):
...
+        8   zebra
+        """
+        if LooseVersion(pyspark.__version__) < LooseVersion("3.0"):
+            raise RuntimeError("tail can be used in PySpark >= 3.0")
+        if not isinstance(n, int):
+            raise TypeError("bad operand type for unary -: '{}'".format(type(n).__name__))
+        if n == 0:
+            return ks.DataFrame(self._internal.with_filter(F.lit(False)))
+        if n < 0:
+            n = len(self) + n
```
Comment (ueshin) [inline databricks/koalas/frame.py:10290]: Maybe:

```py
if n < 0:
    n = len(self) + n
if n <= 0:
    return ks.DataFrame(self._internal.with_filter(F.lit(False)))
```

will be better?

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -4992,6 +4993,62 @@ def items(self) -> Iterable:
...
+            raise TypeError("bad operand type for unary -: '{}'".format(type(n).__name__))
+        if n == 0:
+            return first_series(ks.DataFrame(self._internal.with_filter(F.lit(False))))
+        if n < 0:
+            n = len(self) + n
+        sdf = self._internal.spark_frame
+        rows = sdf.tail(n)
+        new_sdf = default_session().createDataFrame(rows, sdf.schema)
+
+        return first_series(DataFrame(self._internal.with_new_sdf(new_sdf)))
```
Comment (ueshin) [inline databricks/koalas/series.py:5050]: I guess we can reuse the implementation of `DataFrame`?

```py
return first_series(self.to_frame().tail(n=n))
```

File: databricks/koalas/tests/test_dataframe.py [hunk 4]
```diff
@@ -3717,3 +3717,15 @@ def test_iteritems(self):
...
+
+    def test_tail(self):
+        if LooseVersion(pyspark.__version__) >= LooseVersion("3.0"):
+            pdf = pd.DataFrame(range(1000))
+            kdf = ks.from_pandas(pdf)
+
+            self.assert_eq(repr(pdf.tail()), repr(kdf.tail()))
+            self.assert_eq(repr(pdf.tail(10)), repr(kdf.tail(10)))
+            self.assert_eq(repr(pdf.tail(-990)), repr(kdf.tail(-990)))
+            self.assert_eq(repr(pdf.tail(0)), repr(kdf.tail(0)))
```
Comment (ueshin) [inline databricks/koalas/tests/test_dataframe.py:3729]: Why do we using `repr`?

Comment (ueshin) [inline databricks/koalas/tests/test_dataframe.py:3729]: Shall we add more tests, e.g., `tail(1001)`, `tail(-1001)`?

File: databricks/koalas/frame.py (hunk 2 above)
Comment (itholic) [reply databricks/koalas/frame.py:10290]: Ah, seems better. Thanks!

File: databricks/koalas/tests/test_dataframe.py (hunk 4 above)
Comment (itholic) [reply databricks/koalas/tests/test_dataframe.py:3729]: These tests are failed in my local without `repr` like the below.
(Even though they looks same to me.)

```
E   AssertionError: False is not true : DataFrames are not equal:
E
E   Left:
E          0
E   995  995
E   996  996
E   997  997
E   998  998
E   999  999
E   0    int64
E   dtype: object
E
E   Right:
E          0
E   995  995
E   996  996
E   997  997
E   998  998
E   999  999
E   0    int64
E   dtype: object
```

Maybe should it better to use `almost=True` rather `repr` ?

**Comment by itholic** [comment]
Thanks, @ueshin :)

File: databricks/koalas/tests/test_dataframe.py (hunk 4 above)
Comment (ueshin) [reply databricks/koalas/tests/test_dataframe.py]: Maybe the column name of pdf is numeric instead of string.

Comment (itholic) [reply databricks/koalas/tests/test_dataframe.py]: @ueshin Oh, It makes sense. Thanks.

**Pull request merged**